### PR TITLE
add support for refactor log script files

### PR DESCRIPTION
**Pull request by knoxi**
added support for .refactorlog files handled by the original `.sqlproj`

#45

### CODE REVIEW

**Comment by ErikEJ** [comment]
Please remove unneeded whitespace changes

File: src/DacpacTool/Program.cs [hunk 1]
```diff
@@ -94,7 +96,7 @@ private static int BuildDacpac(BuildOptions options)
             }
 
             // Save the package to disk
-            packageBuilder.SaveToDisk(options.Output);
+            packageBuilder.SaveToDisk(options.Output, new PackageOptions() { RefactorLogPath = options.RefactorLog.FullName });
```
Comment (ErikEJ) [inline src/DacpacTool/Program.cs:99]: Add null check before getting .FullName

File: test/TestProjectWithPrePost/RefactorLog/TestProjectWithPrePost.refactorlog [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (jmezach) [inline test/TestProjectWithPrePost/RefactorLog/TestProjectWithPrePost.refactorlog:1]: I see we've added this file, but don't we need to add it as a `<RefactorLog>` item to the `.csproj`. Then we can build the project and verify that it is embedded in the `.dacpac` as expected.

File: src/DacpacTool/PackageBuilder.cs [hunk 3]
```diff
@@ -102,13 +102,12 @@ public bool ValidateModel()
...
         }
 
-        public void SaveToDisk(FileInfo outputFile)
+        public void SaveToDisk(FileInfo outputFile, PackageOptions packageOptions = null)
         {
             // Ensure that the model has been created and metadata has been set
             EnsureModelCreated();
             EnsureModelValidated();
             EnsureMetadataCreated();
-
```
Comment (ErikEJ) [inline src/DacpacTool/PackageBuilder.cs:111]: Revert

File: src/DacpacTool/Program.cs [hunk 4]
```diff
@@ -129,7 +131,8 @@ private static int DeployDacpac(DeployOptions options)
                 }
 
                 deployer.UseTargetServer(options.TargetServerName);
-                
+
```
Comment (ErikEJ) [inline src/DacpacTool/Program.cs:134]: Revert

File: src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets [hunk 5]
```diff
@@ -8,7 +8,6 @@
   <!--
     Set LanguageTargets to Microsoft.Common.targets for any project that the SDK won't (.proj, .noproj, etc)
     https://github.com/dotnet/sdk/blob/50ddfbb91be94d068514e8f4b0ce1052156364a0/src/Tasks/Microsoft.NET.Build.Tasks/sdk/Sdk.targets#L28
-
```
Comment (ErikEJ) [inline src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets:11]: Revert removed line breaks (?)

Comment (ErikEJ) [reply src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets:11]: Sorry, could not see it was whitespace removals...

File: src/DacpacTool/Program.cs (hunk 1 above)
Comment (knoxi) [reply src/DacpacTool/Program.cs]: done

File: test/TestProjectWithPrePost/RefactorLog/TestProjectWithPrePost.refactorlog (hunk 2 above)
Comment (knoxi) [reply test/TestProjectWithPrePost/RefactorLog/TestProjectWithPrePost.refactorlog:1]: I've added the file in the test projects file. Is this enough to test it?

File: src/DacpacTool/PackageBuilder.cs (hunk 3 above)
Comment (knoxi) [reply src/DacpacTool/PackageBuilder.cs]: done

File: src/DacpacTool/Program.cs (hunk 4 above)
Comment (knoxi) [reply src/DacpacTool/Program.cs]: done

File: src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets (hunk 5 above)
Comment (knoxi) [reply src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets]: done

**Comment by knoxi** [comment]
Sorry for the unnessecary code changes with whitespaces.
It's quite hard to work with VS or VSCode and don't format code while saving. It takes me almost longer to revert the unwanted changes then the feature itself.

**Comment by jmezach** [comment]
Perhaps we should add an `.editorconfig` file so that we get a consistent code style.

File: test/TestProjectWithPrePost/RefactorLog/TestProjectWithPrePost.refactorlog (hunk 2 above)
Comment (jmezach) [reply test/TestProjectWithPrePost/RefactorLog/TestProjectWithPrePost.refactorlog:1]: We're not currently building this project as part of CI. Perhaps you can do a `dotnet build` on the TestProjectWihPrePost and then crack open the resulting `.dacpac`  and check whether it includes the refactor.xml file.

Comment (knoxi) [reply test/TestProjectWithPrePost/RefactorLog/TestProjectWithPrePost.refactorlog:1]: Verified, the dacpac file contains the `refactor.xml`
image

Comment (jmezach) [reply test/TestProjectWithPrePost/RefactorLog/TestProjectWithPrePost.refactorlog:1]: Cool!

**Comment by jmezach** [comment]
LGTM! @ErikEJ Are you okay with this as well?

**Comment by ErikEJ** [comment]
LGTM :shipit:

**Pull request merged**